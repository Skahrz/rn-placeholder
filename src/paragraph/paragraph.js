import React from 'react';
import PropTypes from 'prop-types';
import { Line } from '../line/line';
import { Placeholder } from '../placeholder/placeholder';

export class Paragraph extends React.Component {
  constructor(props) {
    super(props);

    this.handleReadyRender = this.handleReadyRender.bind(this);
  }

  handleReadyRender() {
    const { children } = this.props;

    return children;
  }

  render() {
    const {
      lineNumber,
      textSize,
      color,
      width,
      lastLineWidth,
      firstLineWidth,
      children,
      ...props
    } = this.props;

    const lines = [];

    for (let i = 0; i < lineNumber; i++) {
      if (i === 0) {
        lines.push(<Line textSize={textSize} color={color} width={firstLineWidth} key={i} />);
      } else if (i === lineNumber - 1) {
        lines.push(<Line textSize={textSize} color={color} width={lastLineWidth} key={i} />);
      } else {
        lines.push(<Line textSize={textSize} color={color} width={width} key={i} />);
      }
    }

    return (
      <Placeholder {...props} whenReadyRender={this.handleReadyRender}>
        {lines}
      </Placeholder>
    );
  }
}

Paragraph.propTypes = {
  lineNumber: PropTypes.number.isRequired,
  textSize: PropTypes.number,
  color: PropTypes.string,
  width: PropTypes.string,
  lastLineWidth: PropTypes.string,
  firstLineWidth: PropTypes.string,
};

Paragraph.defaultProps = {
  textSize: 12,
  color: '#efefef',
  width: '100%',
  lastLineWidth: '100%',
  firstLineWidth: '100%',
};
