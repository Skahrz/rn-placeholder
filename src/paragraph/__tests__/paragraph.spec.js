import React from 'react';
import { shallow } from 'enzyme';
import { Paragraph } from '../paragraph';

describe('Paragraph', () => {
  let props;

  const getWrapper = () => shallow(<Paragraph {...props} />);

  beforeEach(() => {
    props = {
      lineNumber: 10,
      textSize: 14,
      color: 'green',
      width: '80%',
      lastLineWidth: '40%',
      firstLineWidth: '30%',
    };
  });

  it('should match snapshot', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
