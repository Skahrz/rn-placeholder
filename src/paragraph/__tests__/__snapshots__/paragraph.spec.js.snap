// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should match snapshot 1`] = `
<Placeholder>
  <Line
    color="green"
    key="0"
    textSize={14}
    width="30%"
  />
  <Line
    color="green"
    key="1"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="2"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="3"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="4"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="5"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="6"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="7"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="8"
    textSize={14}
    width="80%"
  />
  <Line
    color="green"
    key="9"
    textSize={14}
    width="40%"
  />
</Placeholder>
`;
